packages/flutter/lib/src/semantics/semantics_event.dart

```diff
@@ -93,13 +93,18 @@ abstract class SemanticsEvent {
 /// [1]: https://developer.android.com/reference/android/view/View#announceForAccessibility(java.lang.CharSequence)
 ///
 class AnnounceSemanticsEvent extends SemanticsEvent {
-  /// Constructs an event that triggers an announcement by the platform.
+  /// Constructs an event that triggers an announcement by the platform
+  /// on the implicit view
   const AnnounceSemanticsEvent(
     this.message,
     this.textDirection, {
     this.assertiveness = Assertiveness.polite,
+    this.viewId = 0,
```

Comment:
The embedder can use a different ID than `0` for the implicit view ID. This could cause problems. Could we make this arg null by default, make `getDataMap` omit `viewId` if it is `null`, and update `SemanticsService.announce` to use `viewId: PlatformDispatcher.instance.implicitView?.viewId`?